dataflow/gpu-examples/conftest.py

```diff
@@ -0,0 +1,508 @@
+                "container",
+                "images",
+                "delete",
+                f"gcr.io/{project}/{image_name}:{UUID}",
+                f"--project={project}",
+                "--force-delete-tags",
+                "--quiet",
+            ]
+            logging.info(f"{cmd}")
+            subprocess.run(cmd, check=True)
+            logging.info(f"Deleted image: gcr.io/{project}/{image_name}:{UUID}")
+
+    @staticmethod
+    def dataflow_jobs_list(
+        project: str = PROJECT, page_size: int = 30
+    ) -> Iterable[dict]:
+        from googleapiclient.discovery import build
+
+        dataflow = build("dataflow", "v1b3")
+
+        response = {"nextPageToken": None}
+        while "nextPageToken" in response:
+            # For more info see:
+            #   https://cloud.google.com/dataflow/docs/reference/rest/v1b3/projects.jobs/list
+            request = (
+                dataflow.projects()
+                .jobs()
+                .list(
+                    projectId=project,
+                    pageToken=response["nextPageToken"],
+                    pageSize=page_size,
+                )
+            )
+            response = request.execute()
+            for job in response["jobs"]:
+                yield job
+
+    @staticmethod
+    def dataflow_jobs_get(
+        job_id: Optional[str] = None,
+        job_name: Optional[str] = None,
+        project: str = PROJECT,
+        list_page_size: int = 30,
+    ) -> Optional[Dict[str, Any]]:
+        from googleapiclient.discovery import build
+
+        dataflow = build("dataflow", "v1b3")
+
+        if job_id:
+            # For more info see:
+            #   https://cloud.google.com/dataflow/docs/reference/rest/v1b3/projects.jobs/get
+            request = (
+                dataflow.projects()
+                .jobs()
+                .get(
+                    projectId=project,
+                    jobId=job_id,
+                    view="JOB_VIEW_SUMMARY",
+                )
+            )
+            # If the job is not found, this throws an HttpError exception.
+            job = request.execute()
+            logging.info(f"Found Dataflow job: {job}")
+            return job
+
+        elif job_name:
+            for job in Utils.dataflow_jobs_list(project, list_page_size):
+                if job["name"] == job_name:
+                    logging.info(f"Found Dataflow job: {job}")
+                    return job
+            raise ValueError(f"Dataflow job not found: job_name={job_name}")
+
+        else:
+            raise ValueError("must specify either `job_id` or `job_name`")
+
+    @staticmethod
+    def dataflow_jobs_wait(
+        job_id: Optional[str] = None,
+        job_name: Optional[str] = None,
+        project: str = PROJECT,
+        region: str = REGION,
+        until_status: str = "JOB_STATE_DONE",
+        timeout_sec: str = 30 * 60,
+        poll_interval_sec: int = 60,
+        list_page_size: int = 100,
```

Comment:
Non blocking nit - what do you think about these also being constants like the other `RETRY_MAX_TIME`? Might help make them more visible for long term maintenance.

Reply:
Sure, sounds good.

On another note, the `RETRY_MAX_TIME` constant is actually not used anymore so I deleted it as well :)